### DISCUSSION TITLE
configure the go runtime to honor the kubernetes memory and cpu resource limits?

### DISCUSSION THREAD

**Discussion by rgl** [rqlite/rqlite#1804] [Q&A]
configure the go runtime to honor the kubernetes memory and cpu resource limits?

like explained at https://www.riverphillips.dev/blog/go-cfs/

with something like https://github.com/rgl/hello-etcd/commit/366b5aba92d69e66bf0da7186fd1e0361320a42b

**Comment by otoolep** [comment]
Sure, go for it. :-)

I say this because this is something that is done outside of rqlite -- you set environment variables at launch time. This is not something that requires changes to rqlite source code.

cc @jtackaberry in case he has any comment for his Helm charts at https://github.com/rqlite/helm-charts

**Comment by otoolep** [comment]
Closing because this is not an rqlite thing, instead it's something anyone can do now if they wish. Just set the environment variables `GOMAXPROCS` and `GOMEMLIMIT` as you want.

**Comment by jtackaberry** [comment]
Interesting idea actually.  It seems pretty straightforward -- even obvious -- but I'll want to research this a bit further before making it the default behavior of the chart.

Meanwhile I've opened https://github.com/rqlite/helm-charts/issues/27 to track it.

**Comment by otoolep** [comment]
I also wrote about something similar in rqlite docs:

https://rqlite.io/docs/guides/performance/#memory-usage
